### DISCUSSION TITLE
nodeEvents

### DISCUSSION THREAD

**Discussion by mattiasb** [argoproj/argo-workflows#9450] [Q&A] [answered]
Hi!

I'd like to understand what `nodeEvents` are. The only explanation I have found are these:

> NodeEvents configures how node events are emitted

— [from the source](https://github.com/argoproj/argo-workflows/blob/master/config/config.go#L25-L26)

> Whether or not to emit events on node completion. These can take a up a lot of space in
> k8s (typically etcd) resulting in errors when trying to create new events:
> "Unable to create audit event: etcdserver: mvcc: database space exceeded"
> This config item allows you to disable this.
> (since v2.9)

— [from docs/workflow-controller-configmap.yaml](https://github.com/argoproj/argo-workflows/blob/master/docs/workflow-controller-configmap.yaml#L33-L39)

The background is that we, last Friday, experienced a bunch of issues related to `etcd` going bananas taking up 50+GB of memory and getting OOM killed a bunch. Looking to mitigate these issues going forward we found the `nodeEvents` setting that we're considering turning off. The issue is, we have no way of knowing what functionality is disabled by disabling `nodeEvents` except for testing. And inspecting the new behavior of Argo.

Could someone more knowledgeable than me please explain what these `nodeEvents` are used for so that we can make an informed decision? :)

Kind regards
Mattias

**Comment by terrytangyuan** [comment, marked as answer]
These are just K8s events to report "NodeSucceeded", "NodeFailed", etc. for individual nodes in workflows. You should be able to disable this safely if you don't have any apps that consume these node events.

**Reply by mattiasb** [reply]
> These are just K8s events to report "NodeSucceeded", "NodeFailed", etc. for individual nodes in 
> workflows. You should be able to disable this safely if you don't have any apps that consume these 
> node events.

Gotcha! And nothing in Argo is using them either?

**Reply by terrytangyuan** [reply]
Correct

**Reply by mattiasb** [reply]
Thanks a lot! ❤️
